backend/controllers/likedvideos.controller.js

```diff
@@ -0,0 +1,63 @@
+import likedVideos from '../models/likedvideos.model.js'
+
+//@route   GET api/likedvideos
+//@desc    Get All Liked Videos
+//@access  Private
+const getAllLikedVideos = async (req, res) => {
+    try {
+        const id = req.user
+        const allvideos = await likedVideos.findOne({ user: id })
+        res.status(200).send(allvideos)
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
+    }
+}
+
+//@route   POST api/likedvideos
+//@desc    Add Video as LikedVideo
+//@access  Private
+
+const addInLikedVideo = async (req, res) => {
+    try {
+        const id = req.user;
+        const { videoID } = req.body
+        const videos = await likedVideos.findOne({ user: id })
+        const isPresent = videos.likedvideos.some((v) => v === videoID)
+        if (isPresent) {
+            return res.status(400).json({ success: false, message: 'Already liked' })
+        }
+        const updatedVideos = [...videos.likedvideos, videoID]
+        const updatedDetails = {
+            user: id,
+            likedvideos: updatedVideos
+        }
+        const addVideo = await likedVideos.findOneAndUpdate({ _id: videos._id }, { $set: updatedDetails }, { new: true })
+        res.status(200).send(addVideo)
+
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
+    }
+}
+
+//@route   DELETE api/likedvideos/:videoID
+//@desc    Remove Video from Liked
+//@access  Private
+
+const removeFromLiked = async (req, res) => {
+    try {
+        const id = req.user;
+        const { videoID } = req.params
+        const videos = await likedVideos.findOne({ user: id })
```

Comment:
Null check required. if there is no videos variable, the filter function would give an error.

Reply:
likedVideos is model which has a field likedvideos which is of type Array. Even if it has no videos it won't cause any problem. So null check is not required here :)